nixos/modules/services/backup/bupstash.nix

```
@@ -0,0 +1,556 @@
+      };
+
+      access = mkOption {
+        type = listOf (
+          submodule {
+            options = {
+              repo = mkOption {
+                type = str;
+                description = "Name of the repository.";
+              };
+
+              keys = mkOption {
+                type = listOf str;
+                default = [ ];
+                description = "List of ssh keys that can be used to access this repository.";
+              };
+
+              allowed = mkOption {
+                type = listOf (
+                  enum [
+                    "init"
+                    "put"
+                    "list"
+                    "get"
+                    "remove"
+                    "gc"
+                    "sync"
+                  ]
+                );
+                default = [ ];
+                description = ''
+                  List of allowed bupstash actions through the ssh connection.
+                  If none are specified then all actions are authorized.
+                '';
+              };
+            };
+          }
+        );
+        default = { };
+        description = "List of ssh keys with access to the repository.";
+      };
+
+      home = mkOption {
+        type = passwdEntry path;
+        default = "/var/lib/${defaultRepositoryUser}";
+        description = "The path where the repositories will be stored.";
+      };
+
+      restrictCommands = mkOption {
+        type = bool;
+        default = true;
+        description = "If set to true, restricts the commands that can be run through ssh by the user to `bupstash ...`.";
+      };
+    };
+
+    jobs = mkOption {
+      type = attrsOf (
+        submodule (
+          { name, ... }:
+          {
+            options = {
+              paths = mkOption {
+                type = listOf path;
+                default = [ ];
+                description = "List of paths to backup.";
+              };
+
+              excludes = mkOption {
+                type = listOf str;
+                default = [ ];
+                description = "List of patterns to exlude from the backup.";
+              };
+
+              command = mkOption {
+                type = nullOr str;
+                default = null;
+                description = "Command used for getting the backup data.";
+              };
+
+              key = mkOption {
+                type = nullOr path;
+                default = null;
+                description = "Path to the key that will be used to encrypt the data.";
+              };
+
+              keyCommand = mkOption {
+                type = nullOr str;
+                default = null;
+                description = ''
+                  A command to run that must print the key data, can be used instead of `job.key`
+                  to fetch the key from arbitrary locations such as the network or other secret storage.
```

```suggestion
                  Run a command that prints the key to stdout. Mutually exclusive with the `key` option.
                  Useful to retrieve the key from password managers.
```

The structure (contract, limitations, example) is good, the wording of the sentences could be simpler and more on point.

Wondering if we can reference other options through a link?